custom_components/elasticsearch/es_doc_creator.py

```diff
@@ -157,33 +157,53 @@ def _state_to_entity_details(self, state: State) -> dict:
         """
         entity_details = self._entity_details.async_get(state.entity_id)
 
-        additions = {}
-
-        if entity_details is not None:
-            additions["device"] = {}
-
-            if entity_details.device:
-                additions["device"]["id"] = entity_details.device.id
-                additions["device"]["name"] = entity_details.device.name
+        if entity_details is None:
+            return {}
+
+        entity = entity_details.entity
+        entity_capabilities = entity.capabilities or {}
+        entity_area = entity_details.entity_area
+
+        entity_additions = {
+            "labels": entity_details.entity_label,
+            "name": entity.name or entity.original_name,
+            "friendly_name": state.name,
+            "platform": entity.platform,
+            "unit_of_measurement": str(entity.unit_of_measurement),
+            "area.id": entity_area.id if entity_area else None,
+            "area.name": entity_area.name if entity_area else None,
+            "class": entity_capabilities.get("state_class"),
```

Comment:
**question** Why did you choose `class` over `state_class` for the property name?

Reply:
value is technically state.value so I was thinking that just class might make sense here, I think in the other PR i lift device_class to be device.class. Happy to just call it state_class though.

Reply:
I slightly prefer `state_class`. Leaving this as `entity.class` gives me the impression that this describes the class of the `entity`, as opposed to the entity's `state`.

Reply:
are you okay with me merging this PR and making new open issues for this and moving the Area and I'll incorporate those into my next PR?

Reply:
Yes that would be fine with me